typescript/api/services/AgendaQueueService.ts

```diff
@@ -185,6 +186,24 @@ export module Services {
 
       }
 
+      /**
+       * 
+       * There are significant slowdowns with Agenda when the jobs collection grows large. This function moves all completed jobs to a history collection so we still have the data but it doesn't affect the peformance of the job processor. 
+       * 
+       * @param job 
+       */
+      public async moveCompletedJobsToHistory(job:any) {
+        const dbManager = User.getDatastore().manager;
+        const collectionName = _.get(sails.config.agendaQueue, 'collection', 'agendaJobs');
+        await dbManager.collection(collectionName).find({ nextRunAt: null }).forEach(async (doc) => {
+          await dbManager.collection(`${collectionName}History`).insertOne(doc);
+          await dbManager.collection(collectionName).deleteOne({ _id: doc._id });
+        });
```

Comment:
Won't this load all the documents only to write them back?

I don't know how many documents there will be, so this might not matter, but iss there a way to do this entirely on the mongodb side?

Reply:
Yes it does. I did look at using a MongoDB aggregation pipeline but I had concerns about concurrency. It should only be a few dozen records at a time at most. Will backlog a ticket to make it more efficient if we find it isn't enough.